internal/ipc/server.go

```diff
@@ -133,35 +144,37 @@ func (ipc *Server) Start() error {
 		wg.Wait()
 	}()
 
-	var retErr error
-	for err := range errsc {
-		if err != nil && retErr == nil {
-			retErr = err
-		}
-	}
-	return retErr
+	return nil
 }
 
-func (ipc *Server) Close() error {
+func (ipc *Server) Shutdown() error {
 	select {
-	case <-ipc.ctx.Done():
+	case <-ipc.donec:
+		return nil
 	default:
+		close(ipc.donec)
 		ipc.cancel()
+		return nil
 	}
-	return nil
 }
 
 func (ipc *Server) Wait() error {
-	ipc.wg.Wait()
-	return nil
+	var retErr error
+	for err := range ipc.errsc {
+		if err != nil && retErr == nil {
+			retErr = err
```

Comment:
We could at least wrap the error messages. Might loose the error type, but you're already at risk of that anyway.

Reply:
This is just a funneling of returned errors, so the wrapping was already performed.